### PR TITLE
[FLINK-19906][table-planner] Fix incorrect result when compare two binary fields

### PR DESCRIPTION
**Pull request by wangxlong**
## What is the purpose of the change

Fix incorrect result when compare two binary fields

## Brief change log

Add SqlFunctionUtils#byteArrayCompare to compare byte arrays.

## Verifying this change

This change added tests and can be verified as follows:
ScalarOperatorsTest#testCompareOperator

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no / don't know)
  - The runtime per-record code paths (performance sensitive): (no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no / don't know)
  - The S3 file system connector: (no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wangxlong** [comment]
@flinkbot run azure

**Review by wuchong — COMMENTED** [review]
cc @danny0405 , could you help to review this?

**Comment by danny0405** [comment]
Sure, let me take a look ~

File: flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/ScalarOperatorsTest.scala [hunk 1]
```diff
@@ -62,6 +62,40 @@ class ScalarOperatorsTest extends ScalarOperatorsTestBase {
     )
   }
 
+  @Test
+  def testCompareOperator(): Unit= {
+
+    // f18 and f19 have same length.
+    testSqlApi(
+      "f18 > f19",
+      "true")
```
Comment (danny0405) [inline flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/ScalarOperatorsTest.scala:71]: We may also need to add some test in `SqlExpressionTest.testComparisonFunctions`

File: flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/functions/SqlFunctionUtils.java [hunk 2]
```diff
@@ -1116,4 +1116,26 @@ public static float struncate(float b0) {
...
+	 * The result is 1 if array1 is great than array2,
+	 * -1 if array1 is less than array2 and 0 if array1 is equal array2.
+	 * Note: Currently, this is used in {@code ScalarOperatorGens}
+	 * for comparing two fields of binary or varbinary type.
+	 *
+	 * @param array1 byte array1 to be compared.
+	 * @param array2 byte array2 to be compared.
+	 * @return Target value.
+	 */
+	public static int byteArrayCompare(byte[] array1, byte[] array2) {
```
Comment (danny0405) [inline flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/functions/SqlFunctionUtils.java:1131]:
```java
/**
	 * Compares two byte arrays in lexicographical order.
	 * 
     * <p>The result is
     * positive if {@code array1} is great than {@code array2},
	 * negative if {@code array1} is less than {@code array2} and
     * 0 if {@code array1} is equal to {@code array2}.
	 * 
     * <p>Note: Currently, this is used in {@code ScalarOperatorGens}
	 * for comparing two fields of binary or varbinary type.
	 *
	 * @param array1 byte array to compare.
	 * @param array2 byte array to compare.
	 * @return an Integer indicating which one is bigger
	 */
```

**Review by danny0405 — COMMENTED** [review]
Thanks @wangxlong , the code looks good, i have left some minor comments.

**Comment by wangxlong** [comment]
@danny0405 Thanks for your reivew. Updated~

**Review by danny0405 — APPROVED** [review]
+1, LGTM.

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**